src/network/generator/main.c

```diff
@@ -81,33 +82,32 @@ static int link_save(Link *link, const char *dest_dir) {
         return 0;
 }
 
-static int context_save(Context *context) {
+static int context_save(Context *context, const char *dest_dir) {
         Network *network;
         NetDev *netdev;
         Link *link;
-        int k, r;
-        const char *p;
+        int k, r = 0;
 
-        p = prefix_roota(arg_root, NETWORKD_UNIT_DIRECTORY);
-
-        r = mkdir_p(p, 0755);
-        if (r < 0)
-                return log_error_errno(r, "Failed to create directory " NETWORKD_UNIT_DIRECTORY ": %m");
+        if (access(dest_dir, F_OK) < 0) {
```

Comment:
hmm?

mkdir_p() is a NOP if the dir already exists. it just creates what is missing. i.e. it's "mkdir -p" but in C